lib/meow_nx/op/selection.ex

```diff
@@ -44,4 +44,35 @@ defmodule MeowNx.Op.Selection do
 
     {genomes, fitness}
   end
+
+  def natural(n) do
+    # TODO: support percentage, something like {:fraction, 0.2}
+    # TODO: validate that n is not larget than population's
+    opts = [n: n]
+
+    %Op{
+      name: "[Nx] Selection natural",
+      requires_fitness: true,
+      invalidates_fitness: false,
+      impl: fn population, _ctx ->
+        Op.map_genomes_and_fitness(population, fn genomes, fitness ->
+          Nx.Defn.jit(&natural_impl(&1, &2, opts), [genomes, fitness], compiler: EXLA)
+        end)
+      end
+    }
+  end
+
+  defnp natural_impl(genomes, fitness, opts \\ []) do
+    final_n = opts[:n]
+    {n, _} = Nx.shape(genomes)
+
+    best_fitness_idx = Nx.argsort(fitness, comparator: :desc)[0..(final_n - 1)]
```

Comment:
I think it would be slightly cleaner if you moved `Nx.argsort(fitness, comparator: :desc)` into a separate variable :>